tests/ttnn/unit_tests/operations/test_moreh_dot.py

```diff
@@ -16,84 +16,55 @@
+        torch_other = torch.randint(-2, 3, input_shape, dtype=cpu_dtype)
     else:
-        input = torch.rand(input_shape, dtype=torch.bfloat16)
-        other = torch.rand(other_shape, dtype=torch.bfloat16)
-        output = torch.rand(output_shape, dtype=torch.bfloat16)
+        torch_input = torch.rand(input_shape, dtype=cpu_dtype)
+        torch_other = torch.rand(input_shape, dtype=cpu_dtype)
 
-    # inputs must be of type bfloat16 or bfloat8
-    tt_input = create_tt_tensor(input, cpu_layout, ttnn.bfloat16).pad_to_tile(float(1)).to(npu_layout).to(device)
-    tt_other = create_tt_tensor(other, cpu_layout, ttnn.bfloat16).pad_to_tile(float("nan")).to(npu_layout).to(device)
-    tt_output = create_tt_tensor(output, cpu_layout, ttnn.bfloat16).pad_to_tile(float("nan")).to(npu_layout).to(device)
+    tt_input = ttnn.from_torch(torch_input, device=device, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT)
+    tt_other = ttnn.from_torch(torch_other, device=device, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT)
 
-    torch_input = input.reshape(-1) if is_1d else input
-    torch_other = other.reshape(-1) if is_1d else other
-
-    # tensors for backward
-    output_grad = tt_output_grad = torch_output_grad = tt_input_grad = tt_other_grad = None
-    if require_input_grad or require_other_grad:
-        output_grad = torch.randint(-2, 3, output_shape, dtype=cpu_dtype)
-        tt_output_grad = (
-            create_tt_tensor(output_grad, cpu_layout, npu_dtype).pad_to_tile(float(-1)).to(npu_layout).to(device)
+    if optional_output == False:
+        tt_out = ttnn.operations.moreh.dot(tt_input, tt_other, dtype=npu_dtype)
+    else:
+        torch_optional_output = torch.rand(output_shape, dtype=cpu_dtype)
+        optional_output = ttnn.from_torch(
+            torch_optional_output, device=device, dtype=npu_dtype, layout=ttnn.TILE_LAYOUT
         )
-        torch_output_grad = output_grad[0][0][0][0] if is_1d else output_grad
+        tt_out = ttnn.operations.moreh.dot(tt_input, tt_other, dtype=npu_dtype, output=optional_output)
+    tt_out = ttnn.to_torch(tt_out)
 
-        if require_input_grad:
-            input_grad = torch.full(input_shape, float("nan"), dtype=cpu_dtype)
-            tt_input_grad = (
-                create_tt_tensor(input_grad, cpu_layout, npu_dtype).pad_to_tile(float("nan")).to(npu_layout).to(device)
-            )
+    # torch matmul
+    torch_input = torch.reshape(torch_input, (torch_input.shape[-1],))
+    torch_other = torch.reshape(torch_other, (torch_other.shape[-1],))
+    torch_out = torch.matmul(torch_input, torch_other)
 
-        if require_other_grad:
-            other_grad = torch.full(other_shape, float("nan"), dtype=cpu_dtype)
-            tt_other_grad = (
-                create_tt_tensor(other_grad, cpu_layout, npu_dtype).pad_to_tile(float("nan")).to(npu_layout).to(device)
-            )
+    if optional_output == True:
+        print(tt_out)
+        print(torch_out)
```

Comment:
Remove print()

Reply:
Resolved

Reply:
Please self-review carefully for these kind of mistakes 😭